### ISSUE TITLE
Add method to convert CqlValue to CQL string

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla-rust-driver#814]
This is a feature request, unless you tell me we already have this feature in the Scylla rust driver (if we do, it's hidden pretty well ;-))

I am doing "SELECT *" on some table with some arbitrary schema (e.g., consider writing a tool that needs to work against any table). I am reading rows from it, and each row is a vector of CqlValue - each can be one of many types (Ascii, Boolean, ....). I want to create a new CQL query based on that response, and for that I need to convert CqlValue to a String in **CQL syntax** - for example the number 7 is printed as the String "7", the string "hello" is printed as "\"hello\"", a list is printed like "[1,2,3]", and so on.

Basically, I hoped to have a CqlValue::to_cql() method.

Any hope for such a thing? Or maybe we already have something similar and I missed it?

**Comment by cvybhu** [comment]
> Any hope for such a thing? Or maybe we already have something similar and I missed it?

We don't have anything like it. It could be added, but I think releasing 1.0 takes has the highest priority atm.

> I want to create a new CQL query based on that response, and for that I need to convert CqlValue to a String 

Maybe it'd be possible to do it without converting CqlValues to string? When you receive a CqlValue from one query, you can send it in another query as a bound value. Pasting in a string is risky, because it could lead to CQL injections.

**Comment by nyh** [comment]
Good idea on the bound value! I'll try it. I didn't realize it also takes CqlValue and not a builtin type. Makes sense.

**Comment by nyh** [comment]
> Maybe it'd be possible to do it without converting CqlValues to string? When you receive a CqlValue from one query, you can send it in another query as a bound value. Pasting in a string is risky, because it could lead to CQL injections.

@cvybhu maybe you can help me, I'm a newbie in both Rust and the Scylla rust driver, and struggling to do this. It appears I need to pass to `session.query()` a "ValueList". I tried to pass what I thought was an iterator of CqlValue, but it wasn't accepted as a ValueList. Is it supposed to work?

**Comment by cvybhu** [comment]
The prettiest way to do it would be to implement `ValueList` for a wrapper over the iterator.

So you would have:
```rust
struct ValuesIter<I: Iterator<Item=CqlValue>(I);

impl ValueList for ValuesIter {
....
}
```

**Comment by cvybhu** [comment]
> The prettiest way to do it would be to implement ValueList for a wrapper over the iterator.
So kinda something like:

Ok this was actually bad advice. The `ValuesTrait` can't mutate the implementor, but we have to mutate the iterator to iterate over it, so it won't work.

I think it would be best to just create SerializedValues based on the iterator.

You can use this function:

```rust
/// Takes an iterator of CQL values (things that implement the Value trait) and serializes them into `SerializedValues`.
/// `SerializedValues` implement the `ValueList` trait, so they can represent a list of bound values for a query.
///
/// ```rust
/// // Here's a prepared statement with three bind variables:
/// let prepared = session
///     .prepare("INSERT INTO ks.tab (a, b, c) VALUES (?, ?, ?)")
///     .await
///     .unwrap();
/// // Usually the values for this statement would be passed as a tuple or a slice of values.
/// session.execute(&prepared, (1, 2, 3)).await.unwrap();
/// session
///     .execute(&prepared, [1, 2, 3].as_slice())
///     .await
///     .unwrap();
/// // There's no straightforward way to pass the bound values as an iterator.
/// // To achieve this we can use the `values_from_iterator` function.
/// let values_iter = [1, 2, 3].into_iter();
/// let serialized_values = values_from_iterator(values_iter).unwrap();
/// session.execute(&prepared, serialized_values).await.unwrap();
/// ```
fn values_from_iterator<V: Value>(
    values_iter: impl Iterator<Item = V>,
) -> Result<SerializedValues, SerializeValuesError> {
    let mut serialized_values: SerializedValues = SerializedValues::new();
    for value in values_iter {
        serialized_values.add_value(&value)?
    }
    Ok(serialized_values)
}
```

**Comment by nyh** [comment]
Thanks. I ended up doing something similar - instead of building "SerializedValues" from the CqlValue's I had (I don't know what this "SerializedValues" is...) I build a Vec from them, and used that vec (no even need for as_slice()) as a parameter to the query, and it works.

So you're probably right that the method I asked for in this issue isn't needed for making requests, so maybe this issue should be closed. On the other hand, maybe it's as feature that's worth having just in case - as you said doing it badly can risk "CQL injection" attacks, so better have one implementation that doesn't do it badly.

**Comment by Lorak-mmk** [comment]
Converting something like CqlValue to string for the purpose of injecting it into query string is basically always a mistake, I'm strongly against exposing such footguns.
For other purposes (debugging, logging / tracing, tools like cqlsh) CqlValue already implements `Display`, so can be printed easily. I don't see anything to do here, so I'm closing it.
